tests/test_core/test_new_sequence.py

```diff
@@ -174,6 +175,51 @@ def test_sequence_strip_degenerate(seq, expect):
     assert got == expect
 
 
+def test_add():
+    mt = new_moltype.DNA
+    seq1 = mt.make_seq(seq="AAA")
+    seq2 = mt.make_seq(seq="CCC")
+    got = seq1 + seq2
+    assert got == "AAACCC"
+    assert got.moltype == mt
+
+
+@pytest.mark.xfail(reason="No error when adding sequences")
+def test_add_bad():
+    mt = new_moltype.DNA
+    seq1 = mt.make_seq(seq="AAA")
+    with pytest.raises(new_alphabet.AlphabetError):
+        _ = f"{seq1}s8d3j%31 s-']"
```

Comment:
This should be changed to actually adding (with `+`) the sequence to the bad string. Including a variable in an f-string exercises whether the variable can be formatted into a string - and since we define a `__str__` method for Sequences we do not expect this to fail. 

To ensure add fails for bad sequences we need to add a validity check in the `__add__` method. 

```py 

if not self.moltype.most_degen_alphabet().is_valid(str(other)):
    raise new_alphabet.AlphabetError(f"Invalid sequence characters in other for moltype={self.moltype.label}")
```